src/parser/Lexer.cpp

```diff
@@ -473,42 +473,31 @@ void ErrorHandler::throwError(size_t index, size_t line, size_t col, String* des
     throw * error;
 };
 
-Scanner::ScannerResult::~ScannerResult()
-{
-    if (this->scanner->isPoolEnabled) {
-        if (this->scanner->initialResultMemoryPoolSize < SCANNER_RESULT_POOL_INITIAL_SIZE) {
-            this->scanner->initialResultMemoryPool[this->scanner->initialResultMemoryPoolSize++] = this;
-            return;
-        }
-        this->scanner->resultMemoryPool.push_back(this);
-    }
-}
-
 StringView Scanner::ScannerResult::relatedSource()
 {
     return StringView(scanner->source, this->start, this->end);
 }
 
 Value Scanner::ScannerResult::valueStringLiteralForAST()
 {
-    StringView sv = valueStringLiteral();
+    StringView* sv = valueStringLiteral();
     if (this->plain) {
-        return new SourceStringView(sv);
+        return new SourceStringView(*sv);
     }
-    RELEASE_ASSERT(sv.string() != nullptr);
-    return sv.string();
+    RELEASE_ASSERT(sv->string() != nullptr);
+    return sv->string();
 }
 
-StringView Scanner::ScannerResult::valueStringLiteral()
+StringView* Scanner::ScannerResult::valueStringLiteral()
 {
     if (this->type == Token::KeywordToken && !this->hasKeywordButUseString) {
         AtomicString as = keywordToString(this->scanner->escargotContext, this->valueKeywordKind);
-        return StringView(as.string(), 0, as.string()->length());
+        return new StringView(as.string(), 0, as.string()->length());
```

Comment:
before return, check nullity of ``new`` with ASSERT

Reply:
Same as above.